truss/templates/server/common/truss_server.py

```diff
@@ -131,7 +132,17 @@ def predict(
             body = json.loads(body_raw)
 
         # calls ModelWrapper.__call__, which runs validate, preprocess, predict, and postprocess
-        response: Dict = asyncio.run(model(body, headers=dict(request.headers.items())))
+        response: Union[Dict, Generator] = asyncio.run(
+            model(
+                body,
+                headers={key.lower(): value for key, value in request.headers.items()},
```

Comment:
minor: a transform_keys function is waiting to get out here. Could go into truss/util/data_structures